akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java

```diff
@@ -160,184 +170,116 @@ public int hashCode() {
-      TestProbe.<String>create(testKit.system()).ref(),
-      (s, i, l) -> false,
-      (i) -> Collections.emptySet(),
-      TestProbe.<Optional<Throwable>>create(testKit.system()).ref(),
-      new NoOpEventAdapter<>()
-    );
-  }
+    @Override
+    public CommandHandler<Command, Incremented, State> commandHandler() {
+      return commandHandlerBuilder(State.class)
+          .matchCommand(Increment.class, this::increment)
+          .matchCommand(GetValue.class, this::getValue)
+          .matchCommand(IncrementLater.class, this::incrementLater)
+          .matchCommand(DelayFinished.class, this::delayFinished)
+          .matchCommand(Increment100OnTimeout.class, this::increment100OnTimeout)
+          .matchCommand(Timeout.class, this::timeout)
+          .matchCommand(EmptyEventsListAndThenLog.class, this::emptyEventsListAndThenLog)
+          .matchCommand(StopThenLog.class, this::stopThenLog)
+          .matchCommand(IncrementTwiceAndLog.class, this::incrementTwiceAndLog)
+          .build();
 
-  private Behavior<Command> counter(
-    String persistenceId,
-    Function3<State, Incremented, Long, Boolean> snapshot,
-    ActorRef<Optional<Throwable>> snapshotProbe
-  ) {
-    return counter(persistenceId,
-      testKit.<Pair<State, Incremented>>createTestProbe().ref(),
-      testKit.<String>createTestProbe().ref(),
-      snapshot,
-      e -> Collections.emptySet(),
-      snapshotProbe,
-      new NoOpEventAdapter<>());
-  }
+    }
 
-  private Behavior<Command> counter(
-    String persistentId,
-    ActorRef<Pair<State, Incremented>> eventProbe,
-    ActorRef<String> loggingProbe) {
-    return counter(persistentId, eventProbe, loggingProbe, (s, i, l) -> false, e -> Collections.emptySet(),
-      TestProbe.<Optional<Throwable>>create(testKit.system()).ref(),
-      new NoOpEventAdapter<>()
-    );
-  }
+    private Effect<Incremented, State> increment(State state, Increment command) {
+      return Effect().persist(new Incremented(1));
+    }
 
-  private Behavior<Command> counter(
-    String persistentId,
-    ActorRef<Pair<State, Incremented>> eventProbe,
-    Function3<State, Incremented, Long, Boolean> snapshot) {
-    return counter(persistentId, eventProbe, testKit.<String>createTestProbe().ref(), snapshot, (e) -> Collections.emptySet(),
-      TestProbe.<Optional<Throwable>>create(testKit.system()).ref(), new NoOpEventAdapter<>()
-    );
-  }
+    private Effect<Incremented, State> getValue(State state, GetValue command) {
+      command.replyTo.tell(state);
+      return Effect().none();
+    }
 
-  private static <A> Behavior<Command> counter(
-    String persistentId,
-    ActorRef<Pair<State, Incremented>> eventProbe,
-    ActorRef<String> loggingProbe,
-    Function3<State, Incremented, Long, Boolean> snapshot,
-    Function<Incremented, Set<String>> tagsFunction,
-    ActorRef<Optional<Throwable>> snapshotProbe,
-    EventAdapter<Incremented, A> transformer) {
```

Comment:
Rather than requiring all of the probes and functions to be specified in all cases, specific tests are now overriding methods that let them insert the probes or other behavior that they need to test.